### PR TITLE
309 nox database configuration for vatnumber

### PR DESCRIPTION
**Pull request by anna-naumova25**
Added database configuration for VatNumber;
Added VatNumber type to AllNoxTypes and data seeder for AllNoxTypes.

### CODE REVIEW

File: src/Nox.Types.Abstractions/TypeOptions/VatNumberOptions.cs
Comment (rochar) [inline src/Nox.Types.Abstractions/TypeOptions/VatNumberOptions.cs:1]: the file name should be aligned with the class name VatNumberTypeOptions

File: src/Nox.Types.Abstractions/Enums/NoxType.cs [hunk 1]
```diff
@@ -50,8 +50,8 @@ public enum NoxType : uint
     DateTimeRange = 3837929056,
 
     [CompoundType]
-    [CompoundComponent("Number", typeof(string))]
-    [CompoundComponent("CountryCode2", typeof(string))]
+    [CompoundComponent("VatNumberValue", typeof(string))]
+    [CompoundComponent("CountryCode", typeof(string))]
```
Comment (rochar) [inline src/Nox.Types.Abstractions/Enums/NoxType.cs:54]: we have different CountryCodes, so CountryCode2 would make sense

File: src/Nox.Types.Abstractions/Enums/NoxType.cs [hunk 2]
```diff
@@ -50,8 +50,8 @@ public enum NoxType : uint
     DateTimeRange = 3837929056,
 
     [CompoundType]
-    [CompoundComponent("Number", typeof(string))]
-    [CompoundComponent("CountryCode2", typeof(string))]
+    [CompoundComponent("VatNumberValue", typeof(string))]
```
Comment (rochar) [inline src/Nox.Types.Abstractions/Enums/NoxType.cs:53]: VatNumber only, value is redundant, agree?

File: src/SampleWebApp/Migrations/20230731144811_InitialCreate.cs [hunk 3]
```diff
@@ -17,6 +17,8 @@ protected override void Up(MigrationBuilder migrationBuilder)
                 {
                     Id = table.Column<string>(type: "char(3)", unicode: false, fixedLength: true, maxLength: 3, nullable: false),
                     TextField = table.Column<string>(type: "nvarchar(63)", maxLength: 63, nullable: false),
+                    VatNumberField_VatNumberValue = table.Column<string>(type: "nvarchar(max)", nullable: false),
```
Comment (rochar) [inline src/SampleWebApp/Migrations/20230731144811_InitialCreate.cs:20]: do we have same max size to VatNumber, or can be anything?

**Review by rochar — CHANGES_REQUESTED** [review]
looks godd, a few comments to review

File: src/Nox.Types.EntityFramework/Types/VatNumber/VatNumberDatabaseConfigurator.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public NoxType ForNoxType => NoxType.VatNumber;
+    public virtual bool IsDefault => true;
+
+    public void ConfigureEntityProperty(NoxSolutionCodeGeneratorState noxSolutionCodeGeneratorState, EntityTypeBuilder builder, NoxSimpleTypeDefinition property, Entity entity, bool isKey)
+    {
+        builder
+            .OwnsOne(typeof(VatNumber), property.Name)
+            .Ignore(nameof(VatNumber.Value))
+            .Property(nameof(VatNumber.CountryCode))
```
Comment (rochar) [inline src/Nox.Types.EntityFramework/Types/VatNumber/VatNumberDatabaseConfigurator.cs:17]: needs something like this .IsUnicode(false) .IsFixedLength() .HasMaxLength(2)

File: src/SampleWebApp/Migrations/20230731144811_InitialCreate.cs (hunk 3 above)
Comment (rochar) [reply src/SampleWebApp/Migrations/20230731144811_InitialCreate.cs:20]: lets limit to 64

**Review by rochar — CHANGES_REQUESTED** [review]

File: src/Nox.Types.Abstractions/TypeOptions/VatNumberOptions.cs
Comment (anna-naumova25) [reply src/Nox.Types.Abstractions/TypeOptions/VatNumberOptions.cs:1]: updated

File: src/Nox.Types.Abstractions/Enums/NoxType.cs (hunk 1 above)
Comment (anna-naumova25) [reply src/Nox.Types.Abstractions/Enums/NoxType.cs]: updated

File: src/Nox.Types.Abstractions/Enums/NoxType.cs (hunk 2 above)
Comment (anna-naumova25) [reply src/Nox.Types.Abstractions/Enums/NoxType.cs]: updated

File: src/SampleWebApp/Migrations/20230731144811_InitialCreate.cs [hunk 5]
```diff
@@ -17,6 +17,8 @@ protected override void Up(MigrationBuilder migrationBuilder)
...
+                    VatNumberField_CountryCode = table.Column<string>(type: "nvarchar(max)", nullable: false),
```
[the remark this answers is not in the document]
Comment (anna-naumova25) [reply src/SampleWebApp/Migrations/20230731144811_InitialCreate.cs]: done

**Review by rochar — APPROVED** [review]

**Pull request merged**